src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class Figure implements FigureInformation {
+    private static final Random random = new Random();
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+    private String color = colorSupplier.getRandomColor();
+
+    public abstract Figure createRandomFigure();
+
+    protected int getRandomInt() {
+        return random.nextInt(100);
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+}
```

Comment:
Don't use abstract classes to set behavior for classes. In this case, `Figure` should not be responsible for creating random figures or supplying colors. Consider separating these responsibilities into different classes or methods outside of the `Figure` hierarchy.